numcodecs/zfpy.py

```diff
@@ -54,8 +55,16 @@ def __init__(
 
         def encode(self, buf):
 
-            # normalise inputs
-            buf = ensure_contiguous_ndarray(buf)
+            # not flatten c-order array and raise exception for f-order array
+            if not isinstance(buf, np.ndarray):
+                raise TypeError("The zfp codec does not support none numpy arrays."
```

Comment:
What other type would you expect here? I thought all Zarr array data would come into numcodecs as numpy arrays?

Reply:
I would like to repeat my answers here, I added test_err_encode_list, which caused buf.flags not raise a correct error. So I added check the instance first. Do you have any different recommendation?